src/nvim/path.c

```diff
@@ -180,6 +180,37 @@ const char *path_next_component(const char *fname)
   return fname;
 }
 
+/// Returns the length of the path head on the current platform.
+/// @return
+///   - 3 on windows
+///   - 1 otherwise
+int path_head_length(void)
+{
+#ifdef WIN32
+  return 3;
+#else
+  return 1;
+#endif
+}
+
+/// Returns true if path begins with characters denoting the head of a path
+/// (e.g. '/' on linux and 'D:' on windows).
+/// @param path The path to be checked.
+/// @return
+///   - True if path begins with a path head
+///   - False otherwise
+bool is_path_head(const char_u *path)
+{
+#ifdef WIN32
+  if (isalpha(path[0]) && path[1] == ':') {
```

Comment:
`return (isalpha(path[0]) && path[1] == ':');`

Reply:
>return (isalpha(path[0]) && path[1] == ':');

Gah, of course. I swear it's because of copy-pasting and I didn't write this on purpose :sweat_smile: .

>path[2] should be forward/backward slash.

I tried this and it makes some fnamemodify oldtests fail.